scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,410 @@
+
+        # generating dcid using the utils/statvar_dcid_generator.py
+        stat_var_dcid = get_stat_var_dcid(stat_var,
+                                          ignore_props=dependent_properties)
+
+        ## overwrite stat_var_dcids from the spec (for existing dcids)
+        if stat_var_dcid in self.features['overwrite_dcids']:
+            stat_var_dcid = self.features['overwrite_dcids'][stat_var_dcid]
+        stat_var['Node'] = 'dcid:' + stat_var_dcid
+
+        #Move the dcid to begining of the dict, uses OrderedDict
+        stat_var = OrderedDict(stat_var)
+        stat_var.move_to_end('Node', last=False)
+        stat_var = json.loads(json.dumps(stat_var))
+
+        #prefix the values if they are not QuantityRanges with dcs:
+        stat_var = self._format_stat_var_node(stat_var)
+
+        #check if the stat_var dict has the mandatory properties
+        if self._isvalid_stat_var(stat_var):
+            return stat_var
+        else:
+            try:
+                raise ValueError
+            except ValueError:
+                logger.critical(
+                    f'One or more mandatory properties of the stat_var is missing. \n Dump of the stat_var:: {stat_var}'
+                )
+
+    # TODO: Rename and re-define this function, since enumSpecializations are used to remove the base class token in the column name and does not affect the stat var defined.
+    def _keep_only_enum_specializations(self):
+        """removes generalization token from column name.
+
+      Example: if the enumSpecicalization is defined as
+      {
+        "Under 6 years": "Under 19 years, Under 18 years",
+        "6 to 18 years": "Under 19 years",
+      }
+      It means that "Under 6 years" and "6 to 18 years" are specializations of "Under 19 years".
+      Hence, if hte token "Under 19 years" and "Under 6 years" occur in the same column name, we drop the token "Under 19 years" from the column
+      """
+        # TODO:The function call takes as an argument a  column name, it removes
+        # the base class from the column name and keeps only the specialization
+        column_map = self.column_map.copy()
+        for column_name, stat_var in column_map.items():
+            part_list = column_name.split(self.delimiter)
+            ## The assumption of using the last token of the column as the specialization holds for ACS Subject tables.
+            # TODO: This assumption might break for other tables, we might need to thing of removing this assumption
+            last_column_token = part_list[-1]
+            if last_column_token in self.features['enumSpecializations']:
+                base_class = self.features['enumSpecializations'][
+                    last_column_token]
+                for base in base_class.split(', '):
+                    if base in part_list:
+                        col_name = self.delimiter.join(part_list[:-1])
+                        try:
+                            del self.column_map[col_name]
```

Comment:
The comment above is not clear enough to indicate an entire column would be dropped.
Can you pls add an example?